src/language-yaml/printer-yaml.js

```diff
@@ -326,18 +323,15 @@ function printNode(node, parentNode, path, options, print) {
       return printBlock(path, print, options);
     }
     case "sequence":
-      return join(hardline, path.map(print, "children"));
+      return join(hardline, print("children"));
     case "sequenceItem":
-      return [
-        "- ",
-        alignWithSpaces(2, !node.content ? "" : path.call(print, "content")),
-      ];
+      return ["- ", alignWithSpaces(2, !node.content ? "" : print("content"))];
     case "mappingKey":
-      return !node.content ? "" : path.call(print, "content");
+      return !node.content ? "" : print("content");
     case "mappingValue":
-      return !node.content ? "" : path.call(print, "content");
+      return !node.content ? "" : print("content");
     case "mapping":
-      return join(hardline, path.map(print, "children"));
+      return join(hardline, print("children"));
```

Comment:
Changes like this look bad to me. `path.map` is more type-safe. Previously we knew for sure that `map` would return an array. Now we can only hope it will do so.

Reply:
I was not sure about this change too, but we can't use `path.map(print, "children")` anymore, we need use `path.map(() => print(), "children")`, because `print` don't accept `path`. This is the reason I made this change.

Reply:
FYI: The old `path.map(print, "children")` is also not good, it send `index` as `args`, I didn't realize this when I add `index` to `AstPath.map`.

Reply:
It wasn't you. It did that from the beginning. Which led to disabled caching, which probably was a good thing. Difficult to tell without performance testing.

Reply:
Oh, I remember it incorrectly, I added the array to callback #9901.

Anyway, if print don't handle `array` automatically, I think we'll need write this in 

1.

```js
path.map(() => print(), "children");
```

2.

```js
for (let index = 0; index < node.children.length; index ++) {
	parts.push(print(["children", index]))
}
```

All less easier than `print("children")`

Reply:
The `index` is on me... #9499

Reply:
I'd really prefer if `path.map(print, "children");` kept working.

Reply:
What if we use `print.array("children")`?

Reply:
> I'd really prefer if path.map(print, "children"); kept working.

If I remove [this line](https://github.com/prettier/prettier/pull/10557/files#diff-9e781c3963d9712c66a4747944b15dbc98d3e504f3b126537a746051946f1131R87), it will still work the old way. But it makes `printGenerically` very suspicious.

Reply:
We should keep supporting the old way anyway. Why make plugin authors change this? What for?